doc/modules/language-guide/pages/motoko-introduction.adoc

```diff
@@ -4,35 +4,42 @@
+Specifically, {proglang} programs are _type sound_ since {proglang} includes a practical, modern type system that checks each one before it executes.
+The {proglang} type system statically checks that each {proglang} program will execute safely, without dynamic type errors, on all possible inputs.
+Consequently, entire classes of common programming pitfalls that are common in other languages, and web programming languages in particular, are ruled out. This includes null reference errors, mis-matched argument or result types, missing field errors and many others.
 
 To execute, {proglang} statically compiles to link:about-this-guide{outfilesuffix}#wasm[WebAssembly], a portable binary format that abstracts cleanly over modern computer hardware, and thus permits its execution broadly on the Internet, and the link:../developers-guide/concepts/what-is-ic{outfilesuffix}#ic-overview[{IC}].
 
 [[pitch-actors]]
-== Each microservice as an _actor_
+== Each canister smart constract as an _actor_
 
-{proglang} provides an *actor-based* programming model to developers to express _server behavior_, including that of _micro services_ on the Internet and the link:../developers-guide/concepts/what-is-ic{outfilesuffix}#ic-overview[{IC}].
+{proglang} provides an *actor-based* programming model to developers to express _services_, including those of canister smart contracts on the link:../developers-guide/concepts/what-is-ic{outfilesuffix}#ic-overview[{IC}].
 
-An actor is similar to an object, but is special in that its isolated state exists _remotely_, and its interactions with the world are _asynchronous_.
+An actor is similar to an object, but is special in that it state is completely isolated, and all its interactions with the world are by _asynchronous_ messaging.
 
 All communication with and between actors involves passing messages asynchronously over the network using the Internet Computer's messaging protocol.
-An actor’s messages are processed in sequence, so state modifications never cause race conditions.
+An actor’s messages are processed in sequence, so state modifications never admit race conditions (unless explicitly allowed by punctuating `await`s).
 
-The underlying computing platform provided by the Internet Computer ensures that each message-based synchronization will reach a close, but in doing so each one may also fail due to the usual reasons that arise in distributed systems, including time out.
+The Internet Computer ensures that each message that is sent receives
+a response. The response is either success with some value, or
+an error. An error can be the explicit rejection of the message by the
+receiving canister, a trap due to an illegal instruction such as division by zero, or a system error due to distribution or resource constraints.
+For example, a system error might be
+the transient or permanent unavailability of the receiver (either because
+the receiving actor is oversubscribed or has been deleted).
```

Comment:
Not a user-visible thing but this paragraph seems to be split into multiple lines randomly (some lines long, others short). Maybe merge this as a single line? If you use vim, you can also select the text and do `gq` to automatically format it based on your `textwidth`.

Reply:
Done